ReactWindows/ReactNative/Common/TaskExtensions.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ReactNative.Common
+{
+    static class TaskExtensions
+    {
+        static async Task<B> SelectMany<A, B>(
+            this Task<A> task,
+            Func<A, Task<B>> function)
+        {
+            A a = await task;
+            return await function(a);
+        }
+
+        public static async Task<V> SelectMany<T, U, V>(this Task<T> source, Func<T, Task<U>> selector, Func<T, U, V> resultSelector)
```

Comment:
I don't think you need both SelectMany methods, let's stick with just what we need for now.